### PR TITLE
Updates and Fixes

### PR DESCRIPTION
**Pull request by jdent02**

### CODE REVIEW

File: utils/util.py [hunk 1]
```diff
@@ -112,6 +112,9 @@ def update_project(_):
             for cls in node_classes:
                 safe_register_class(cls)
 
+        else:
+            logger.debug(f"appleseed: Shader {script.name} did not compile")
```
Comment (dictoon) [inline utils/util.py]: Is there a way to include the error message(s)?

File: translators/objects/mesh.py [hunk 2]
```diff
@@ -278,14 +278,14 @@ def __get_material_mappings(self):
 
         if len(material_slots) > 1:
             for i, m in enumerate(material_slots):
-                if m.material.use_nodes is True:
+                if m.material is not None and m.material.use_nodes is True:
```
Comment (dictoon) [inline translators/objects/mesh.py]: Note that `is True` is redundant.

File: translators/scene.py [hunk 3]
```diff
@@ -529,7 +529,8 @@ def __translate_render_settings(self):
                       'passes': asr_scene_props.renderer_passes,
                       'generic_frame_renderer': {'tile_ordering': asr_scene_props.tile_ordering},
                       'progressive_frame_renderer': {'max_average_spp': asr_scene_props.interactive_max_samples,
-                                                     'max_fps': asr_scene_props.interactive_max_fps},
+                                                     'max_fps': asr_scene_props.interactive_max_fps,
+                                                     'time_limit': asr_scene_props.interactive_max_time},
```
Comment (dictoon) [inline translators/scene.py]: Rename `time_limit` to `max_time` for consistency with the other properties?

File: translators/scene.py [hunk 4]
```diff
@@ -676,6 +676,27 @@ def __translate_frame(self, context):
             window_y_min = height - int(camera_height * self.bl_scene.render.border_max_y + window_shift_y - view_shift_y)
             window_y_max = height - int(camera_height * self.bl_scene.render.border_min_y + window_shift_y - view_shift_y)
 
+            #Check for coordinates outside the render window
```
Comment (dictoon) [inline translators/scene.py]: Add space after `#`, terminate sentence with a period.

File: translators/scene.py [hunk 5]
```diff
@@ -676,6 +676,27 @@ def __translate_frame(self, context):
...
+            if window_x_min < 0:
+                window_x_min = 0
+            elif window_x_min > width:
```
Comment (dictoon) [inline translators/scene.py]: That doesn't look right, for a WxH image, the max cropw window is [0,W-1] x [0,H-1].

I would also suggest introducing a function (if it doesn't already exist) that clamp a value to min and max bounds, i.e. `window_x_max = clamp(window_x_max, 0, width - 1)`.

File: utils/util.py (hunk 1 above)
Comment (jdent02) [reply utils/util.py]: I haven’t found one. There is an error handler for OIIO tex conversion that’s supposed to print to stdout or stderr but it doesn’t seem to do anything.

File: translators/scene.py (hunk 5 above)
Comment (jdent02) [reply translators/scene.py]: With the window position calculations, if a user shifts the through camera view (using Shift-Middle Mouse) one or more of the border sides could end up outside the view window, leading to a window dimension that is larger than the width/height or a negative number.

File: translators/objects/mesh.py (hunk 2 above)
Comment (jdent02) [reply translators/objects/mesh.py]: Fixed

File: translators/scene.py (hunk 3 above)
Comment (jdent02) [reply translators/scene.py]: Can't, that's the name of the appleseed parameter.

File: translators/scene.py (hunk 5 above)
Comment (jdent02) [reply translators/scene.py]: Although a clamp would be an alternative.

File: translators/scene.py (hunk 3 above)
Comment (dictoon) [reply translators/scene.py]: Doh!

**Comment by jdent02** [comment]
Should be fixed.

**Pull request merged**